dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLMemoryStore.cs

```diff
@@ -0,0 +1,462 @@
+    /// <param name="indexingPolicy">The <see cref="IndexingPolicy"/> to use if a collection is created. NOTE that embeddings will be stored in a property named 'embedding'.</param>
+    public AzureCosmosDBNoSQLMemoryStore(
+        CosmosClient cosmosClient,
+        string databaseName,
+        VectorEmbeddingPolicy vectorEmbeddingPolicy,
+        IndexingPolicy indexingPolicy)
+    {
+        if (!vectorEmbeddingPolicy.Embeddings.Any(e => e.Path == "/embedding"))
+        {
+            throw new InvalidOperationException($"""
+                In order for {nameof(GetNearestMatchAsync)} to function, {nameof(vectorEmbeddingPolicy)} should
+                contain an embedding path at /embedding. It's also recommended to include a that path in the
+                {nameof(indexingPolicy)} to improve performance and reduce cost for searches.
+                """);
+        }
+        this._cosmosClient = cosmosClient;
+        this._databaseName = databaseName;
+        this._vectorEmbeddingPolicy = vectorEmbeddingPolicy;
+        this._indexingPolicy = indexingPolicy;
+    }
+
+    /// <inheritdoc/>
+    public async Task CreateCollectionAsync(
+        string collectionName,
+        CancellationToken cancellationToken = default)
+    {
+        var databaseResponse = await this._cosmosClient.CreateDatabaseIfNotExistsAsync(
+            this._databaseName, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+        var containerProperties = new ContainerProperties(collectionName, "/key")
+        {
+            VectorEmbeddingPolicy = this._vectorEmbeddingPolicy,
+            IndexingPolicy = this._indexingPolicy,
+        };
+        var containerResponse = await databaseResponse.Database.CreateContainerIfNotExistsAsync(
+            containerProperties,
+            cancellationToken: cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<string> GetCollectionsAsync(
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        using var feedIterator = this.
+            _cosmosClient
+            .GetDatabase(this._databaseName)
+            .GetContainerQueryIterator<string>("SELECT VALUE(c.id) FROM c");
+
+        while (feedIterator.HasMoreResults)
+        {
+            var next = await feedIterator.ReadNextAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var containerName in next.Resource)
+            {
+                yield return containerName;
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task<bool> DoesCollectionExistAsync(
+        string collectionName,
+        CancellationToken cancellationToken = default)
+    {
+        using var feedIterator = this.
+            _cosmosClient
+            .GetDatabase(this._databaseName)
+            .GetContainerQueryIterator<string>($"SELECT VALUE(c.id) FROM c WHERE c.id = '{collectionName}'");
```

Comment:
Any SQL injection attack concerns here?

Reply:
SQL injection is harder in Cosmos DB, since our SQL syntax doesn't allow modification, but I suppose it might be possible to construct something to disclose extra information, so I've switched all places that use argument data in queries to use parameterized queries.